src/main/java/com/minecolonies/coremod/client/gui/WindowRequestDetail.java

```diff
@@ -0,0 +1,237 @@
+        }
+        else
+        {
+            exampleStackDisplay.setItem(ItemStackUtils.EMPTY);
+        }
+    }
+
+    /**
+     * Called when the GUI has been opened.
+     * Will fill the fields and lists.
+     */
+    @Override
+    public void onOpened()
+    {
+        String[] labels = request.getLongDisplayString().getFormattedText()
+                            .replace("§r", " ")
+                            .replace(": ", ":\n")
+
+                            .split("(?<=\n)");
+
+        final StringBuilder finalLabel = new StringBuilder();
+
+        for (final String s : labels)
+        {
+            finalLabel.append(WordUtils.wrap(s, WRAP_AFTER_X, "\n", true));
+        }
+
+        labels = finalLabel.toString().split("\n");
+        final Box box = findPaneOfTypeByID(BOX_ID_REQUEST, Box.class);
+        int y = Y_OFFSET_EACH_TEXTFIELD;
+        for (final String s : labels)
+        {
+            final Label descriptionLabel = new Label();
+            descriptionLabel.setColor(BLACK, BLACK);
+            descriptionLabel.setLabelText("§r§0" + s);
+            box.addChild(descriptionLabel);
+            descriptionLabel.setPosition(1, y);
+            y += Y_OFFSET_EACH_TEXTFIELD;
+        }
+
+        final ItemIcon exampleStackDisplay = findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_STACK, ItemIcon.class);
+        final List<ItemStack> displayStacks = request.getDisplayStacks();
+
+        if (!displayStacks.isEmpty())
+        {
+            exampleStackDisplay.setItem(displayStacks.get((lifeCount / LIFE_COUNT_DIVIDER) % displayStacks.size()));
+        }
+        else
+        {
+            final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
+            logo.setVisible(true);
+            logo.setImage(request.getDisplayIcon());
+        }
+
+        findPaneOfTypeByID(REQUESTER, Label.class).setLabelText(request.getRequester().getDisplayName(request.getToken()).getFormattedText());
+        final Label targetLabel = findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_LOCATION, Label.class);
+        targetLabel.setLabelText(request.getRequester().getDeliveryLocation().toString());
+
+
+        final ColonyView colony = ColonyManager.getColonyView(colonyId);
+        if(colony == null)
+        {
+            Log.getLogger().warn("---Colony Null in WindowRequestDetail---");
+            return;
+        }
+
+        try
+        {
+            final IRequestResolver resolver = colony.getRequestManager().getResolverForRequest(request.getToken());
+            if(resolver == null)
+            {
+                Log.getLogger().warn("---IRequestResolver Null in WindowRequestDetail---");
+                return;
+            }
+
+            findPaneOfTypeByID(RESOLVER, Label.class).setLabelText("Resolver: " + resolver.getDisplayName(request.getToken()).getFormattedText());
+        }
+        catch(@SuppressWarnings(EXCEPTION_HANDLERS_SHOULD_PRESERVE_THE_ORIGINAL_EXCEPTIONS) final IllegalArgumentException e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)